### ISSUE TITLE
crashes if git source crate installed with --branch

### ISSUE THREAD

**Issue by matthiaskrgr** [nabijaczleweli/cargo-update#64]
So I installed a crate from a git repo with something like
```` cargo install --git <repo> --force --branch <branch>```` which seems to break cargo install update:
````
 RUST_BACKTRACE=1 ./target/debug/cargo-install-update install-update -lag
[...]
tokei             v7.0.0     v7.0.0    No
tree-rs           v0.6.4     v0.6.4    No

thread 'main' panicked at 'called `Result::unwrap()` on an `Err` value: Error { code: -1, klass: 12, message: "unexpected HTTP status code: 400" }', libcore/result.rs:916:5
stack backtrace:
   0: std::sys::unix::backtrace::tracing::imp::unwind_backtrace
             at libstd/sys/unix/backtrace/tracing/gcc_s.rs:49
   1: std::sys_common::backtrace::print
             at libstd/sys_common/backtrace.rs:68
             at libstd/sys_common/backtrace.rs:57
   2: std::panicking::default_hook::{{closure}}
             at libstd/panicking.rs:380
   3: std::panicking::default_hook
             at libstd/panicking.rs:396
   4: std::panicking::rust_panic_with_hook
             at libstd/panicking.rs:576
   5: std::panicking::begin_panic
             at libstd/panicking.rs:537
   6: std::panicking::begin_panic_fmt
             at libstd/panicking.rs:521
   7: rust_begin_unwind
             at libstd/panicking.rs:497
   8: core::panicking::panic_fmt
             at libcore/panicking.rs:71
   9: core::result::unwrap_failed
             at /checkout/src/libcore/macros.rs:23
  10: <core::result::Result<T, E>>::unwrap
             at /checkout/src/libcore/result.rs:782
  11: cargo_update::ops::GitRepoPackage::pull_version
             at ./src/ops/mod.rs:333
  12: cargo_install_update::actual_main
             at src/main.rs:167
  13: cargo_install_update::main
             at src/main.rs:21
  14: std::rt::lang_start::{{closure}}
             at /checkout/src/libstd/rt.rs:74
  15: std::panicking::try::do_call
             at libstd/rt.rs:59
             at libstd/panicking.rs:479
  16: __rust_maybe_catch_panic
             at libpanic_unwind/lib.rs:102
  17: std::rt::lang_start_internal
             at libstd/panicking.rs:458
             at libstd/panic.rs:358
             at libstd/rt.rs:58
  18: std::rt::lang_start
             at /checkout/src/libstd/rt.rs:74
  19: main
  20: __libc_start_main
  21: _start
````

**Comment by matthiaskrgr** [comment]
Alright, the problem is inside ````pull_version()```` we pass a malformed self.url ````<repo>?branch=<branchname>```` into ````git2::build::RepoBuilder::new().clone()````

**Comment by matthiaskrgr** [comment]
You can see it with this debug code
````rust
    pub fn pull_version<P: AsRef<Path>>(&mut self, temp_dir: P) {
        fs::create_dir_all(temp_dir.as_ref()).unwrap();
        let clone_dir = temp_dir.as_ref().join(&self.name);
        let repo = if clone_dir.exists() {
            let mut r = git2::Repository::open(clone_dir);
            if let Ok(ref mut r) = r.as_mut() {
                r.find_remote("origin").and_then(|mut rm| rm.fetch(&["master"], None, None)).unwrap();
            }
            println!("this codebranch");
            r
        } else {
            println!("that codebranch");
            println!("url {:?}", self.url);
            println!("dir {:?}", &clone_dir);


            git2::build::RepoBuilder::new()
                .bare(true)
                .clone(&self.url, &clone_dir)

        };
        println!("end of branches")
        self.newest_id = Some(repo.and_then(|r| r.head().and_then(|h| h.target().ok_or_else(|| GitError::from_str("HEAD not a direct reference")))).unwrap());
    }
````

**Comment by nabijaczleweli** [comment]
Does current `master` work for you? I tested it in my fabricated env but nothing beats a real usecase.

**Comment by matthiaskrgr** [comment]
Looks good here, panic is gone. :) Thanks!

**Comment by nabijaczleweli** [comment]
Released in [`v1.4.1`](https://github.com/nabijaczleweli/cargo-update/releases/tag/v1.4.1)
